### ISSUE TITLE
init model not valid when running the demo code

### ISSUE THREAD

**Issue by tianqiyy** [Boyle-Lab/TRACE#5]
Hi, I directly ran 3rd step of the demo code. However, it always give the following message:
Error: File ./data/E2F1_init_model.txt not valid
I also tried to regenerate the E2F1_init_model.txt file by running the 1st step of demo code, yet it did not solve the problem.
Is that because I must ran the 2nd step to regenerate required seq, slope and count files before doing the 3nd step?

Another question is that why use ./data/E2F1_slope_2.txt but not ./data/E2F1_slope_1.txt in the 3rd step?

**Comment by OuyangNX** [comment]
Hi, it looks like you don't have the model file at that path. Did you cloned the entire TRACE repository including `./data`? Or were the input files in the correct path?  You don't need to regenerate seq, slope and counts file, all files required for demo are all in `./data`.  
`./data/E2F1_slope_2.txt` is used because it is the derivative values that will be used in model training and decoding.

**Comment by tianqiyy** [comment]
I did cloned the entire TRACE repository, and I double checked that all files exist and the package is installed correctly, but it still warns the init model is invalid.

![image](https://user-images.githubusercontent.com/38643012/125494893-84f0145d-1733-42ab-a7bf-2440ceb021c6.png)

**Comment by OuyangNX** [comment]
That's strange, if you change `strcat(hmminitfile, optarg);` to `strcpy(hmminitfile, optarg);`  in `esthmm.c` and re-build, does that work?

**Comment by tianqiyy** [comment]
Still not work after editing the esthmm.c.

Does this package requires the latest version (2.7) of GSL library?

**Comment by tianqiyy** [comment]
As a follow up of the previous question, I currently use GSL (version 2.6), which is the latest version available in the computer cluster environment I am using. Do I need to update to GSL (version 2.7) to allow the program correctly read E2F1_init_model.txt file?

**Comment by OuyangNX** [comment]
No, the program doesn't require the latest version of GSL, and it was actually able to read seq, slope and counts files before reporting error accessing the model file.

**Comment by tianqiyy** [comment]
Hi, I still cot not solve the problem of invalid initial model when running the sample code using the sample files.
At the same time, I regenerated the init model, made seq, slope, count files using my own data and tried to do training and footprinting following the steps of sample code. Again, the warning message still says the init model is not valid.
Similar problem also happens when I asked another person to test the sample code.

I wonder do you have any other suggestions to solve such problem?
